From afb343a97cc354f1156efb1fe4b4470a9a5d5fb9 Mon Sep 17 00:00:00 2001
From: Andrey Butirsky <butirsky@gmail.com>
Date: Tue, 21 Sep 2021 19:11:46 +0300
Subject: [PATCH] [keyboardlayout applet] fix non-working context menu

After refactoring, keyboardLayout plugin is not accessible any more from
root Item object, so we pass the needed info from Action via signal.

BUG: 441239
---
 applets/keyboardlayout/contents/ui/main.qml | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/applets/keyboardlayout/contents/ui/main.qml b/applets/keyboardlayout/contents/ui/main.qml
index 31dee9f16..68956b57b 100644
--- a/applets/keyboardlayout/contents/ui/main.qml
+++ b/applets/keyboardlayout/contents/ui/main.qml
@@ -18,6 +18,12 @@ Item {
                         "kf5/locale/countries/" + name + "/flag.png")
     }
 
+    signal layoutSelected(int layout)
+
+    function actionTriggered(selectedLayout) {
+        layoutSelected(selectedLayout)
+    }
+
     Plasmoid.preferredRepresentation: Plasmoid.compactRepresentation
 
     Plasmoid.compactRepresentation: KeyboardLayoutSwitcher {
@@ -46,10 +52,13 @@ Item {
             }
         }
 
-        function actionTriggered(selectedLayout) {
-            keyboardLayout.layout = selectedLayout
-        }
+        Connections {
+            target: root
 
+            function onLayoutSelected(layout) {
+               keyboardLayout.layout = layout
+            }
+        }
 
         hoverEnabled: true
 
-- 
GitLab

